File: seekbar-compat/src/main/java/app/minimize/com/seek_bar_compat/SeekBarBackgroundDrawable.java
package app.minimize.com.seek_bar_compat;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.ColorFilter;
import android.graphics.Paint;
import android.graphics.drawable.Drawable;

/***
 * A drawable used as a ProgressBackgroundDrawable
 */
public class SeekBarBackgroundDrawable extends Drawable {

    private Paint mPaintLine = new Paint();
    private Paint mPaintBackground = new Paint();
    private float dy;
    private float mPaddingLeft;
    private float mPaddingRight;


    public SeekBarBackgroundDrawable(Context ctx, int lineColor, int backgroundColor, final float paddingLeft, final float paddingRight) {
        mPaddingLeft = paddingLeft;
        mPaddingRight = paddingRight;
        mPaintLine.setColor(lineColor);

        mPaintBackground.setColor(backgroundColor);
        dy = ctx.getResources()
                .getDimension(R.dimen.one_dp);
    }

    @Override
    public void draw(Canvas canvas) {
        canvas.drawRect(getBounds().left,
                getBounds().top,
                getBounds().right,
                getBounds().bottom,
                mPaintBackground);

        canvas.drawRect(getBounds().left+mPaddingLeft,
                getBounds().centerY() - dy / 2,
                getBounds().right-mPaddingRight,
                getBounds().centerY() + dy / 2,
                mPaintLine);
    }


    @Override
    public void setAlpha(final int alpha) {

    }


    @Override
    public void setColorFilter(ColorFilter colorFilter) {

    }

    @Override
    public int getOpacity() {
        return 0;
    }

}

File: seekbar-compat/src/main/java/app/minimize/com/seek_bar_compat/SeekBarBackgroundDrawable.kt
package app.minimize.com.seek_bar_compat

import android.content.Context
import android.graphics.Canvas
import android.graphics.ColorFilter
import android.graphics.Paint
import android.graphics.PixelFormat
import android.graphics.drawable.Drawable

/***
 * A drawable used as a ProgressBackgroundDrawable
 */
class SeekBarBackgroundDrawable(ctx: Context, lineColor: Int, backgroundColor: Int, private val mPaddingLeft: Float, private val mPaddingRight: Float) : Drawable() {

    private val mPaintLine = Paint()
    private val mPaintBackground = Paint()
    private val dy: Float

    init {
        mPaintLine.color = lineColor

        mPaintBackground.color = backgroundColor
        dy = ctx.resources
                .getDimension(R.dimen.one_dp)
    }

    override fun draw(canvas: Canvas) {
        canvas.drawRect(bounds.left.toFloat(),
                bounds.top.toFloat(),
                bounds.right.toFloat(),
                bounds.bottom.toFloat(),
                mPaintBackground)

        canvas.drawRect(bounds.left + mPaddingLeft,
                bounds.centerY() - dy / 2,
                bounds.right - mPaddingRight,
                bounds.centerY() + dy / 2,
                mPaintLine)
    }


    override fun setAlpha(alpha: Int) {

    }


    override fun setColorFilter(colorFilter: ColorFilter?) {

    }

    override fun getOpacity(): Int {
        return PixelFormat.OPAQUE
    }

}

